src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NuGetManagedTemplatePackage.cs

```diff
@@ -235,25 +170,22 @@ public bool IsLocalPackage
 
         public IReadOnlyDictionary<string, string> GetDetails()
         {
-            Dictionary<string, string> details = new Dictionary<string, string>();
-            if (!string.IsNullOrWhiteSpace(Author))
-            {
-                details[AuthorKey] = Author!;
-            }
-            if (!string.IsNullOrWhiteSpace(Owners))
-            {
-                details[OwnersKey] = Owners!;
-            }
-            if (!string.IsNullOrWhiteSpace(Trusted))
-            {
-                details[TrustedKey] = Trusted!;
-            }
-            if (!string.IsNullOrWhiteSpace(NuGetSource))
-            {
-                details[NuGetSourceKey] = NuGetSource!;
-            }
+            var details = new Dictionary<string, string>();
+
+            details.TryAdd(AuthorKey, Author ?? string.Empty, (string entry) => !string.IsNullOrEmpty(entry));
+            details.TryAdd(OwnersKey, Owners ?? string.Empty, (string entry) => !string.IsNullOrEmpty(entry));
+            details.TryAdd(TrustedKey, Trusted ?? string.Empty, (string entry) => !string.IsNullOrEmpty(entry));
+            details.TryAdd(NuGetSourceKey, NuGetSource ?? string.Empty, (string entry) => !string.IsNullOrEmpty(entry));
```

Comment:
Given we need to do coalesce here, i don't see much benefit in changing that code.

Reply:
I think performance wise there isn't a lot of benefit, but I do like that it makes it easier to navigate the file. Having something like this also allows us to coalesce in other parts of the code with this new function.